inngest/src/result.rs

```diff
@@ -36,24 +36,97 @@ impl IntoResponse for SdkResponse {
 }
 
 #[derive(Debug)]
-pub enum InngestError {
+pub enum SimpleError {
```

Comment:
An awful name. But this serves as the "dev facing" error, so the user doesn't have to interact with the control flow error.

Reply:
this probably could just be named `Error`, wdyt?

Reply:
I was initially thinking that
- `RetryAt`
- `NoRetry`
should still be part of `InngestError`.

mainly because they can be used and returned by the users anywhere within the function including steps, and should be propagated up to the handler.

they're really more for signaling to the handler what to respond to the executor, and is not really intended for the user to do anything with them.

ref:
https://www.inngest.com/docs/features/inngest-functions/error-retries/inngest-errors#retry-after-error

Reply:
As for `Basic`, I'm not quite sure what to do with it yet.
It was merely a quick placeholder to put around in the code, but not quite sure if it's actually needed yet.

But maybe we can keep it as a user accessible enum. It's probably good to treat it as a retriable type of error but it just uses the default backoff strategies, whereas `RetryAfter` gives the user more control of how they want to handle retries programatically.

Reply:
Since RetryAt, NoRetry, Basic are all dev facing, I've named that enum `DevError`. The parent error enum, which includes internal errors that the user/developer shouldn't interact with is now named `Error`.